src/rules/variableNameRule.ts

```diff
@@ -29,30 +34,48 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 class VariableNameWalker extends Lint.RuleWalker {
+    private checkFormat: boolean;
+    private banKeywords: boolean;
+
+    constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+        super(sourceFile, options);
+
+        this.banKeywords = this.hasOption(OPTION_BAN_KEYWORDS);
+        // check variable name formatting by default if no options are specified
+        this.checkFormat = !this.banKeywords || this.hasOption(OPTION_CHECK_FORMAT);
+    }
+
     public visitBindingElement(node: ts.BindingElement) {
         if (node.name.kind === ts.SyntaxKind.Identifier) {
-            this.handleVariableName(<ts.Identifier> node.name);
+            const identifier = node.name as ts.Identifier;
```

Comment:
I'm not sure how much type guards could help. They don't work on dotted names so the code probably would be just about as as verbose